loki/bulk/scheduler.py

```diff
@@ -638,9 +650,16 @@ def process(self, transformation, reverse=False, item_filter=SubroutineItem, use
                     # Use entry from item_map to ensure the original item is used in transformation
                     _item = self.item_map[item.name]
 
+                    # Pick out the IR node to which to apply the transformation
+                    # TODO: should this become an Item property?
```

Comment:
[no action] Yes, it should, but it requires the SGraph development to make this behaviour neat. So good for now.